xds/internal/resolver/serviceconfig.go

```diff
@@ -156,7 +159,16 @@ func (cs *configSelector) SelectConfig(rpcInfo iresolver.RPCInfo) (*iresolver.RP
 		}
 	}
 	if rt == nil || rt.clusters == nil {
-		return nil, errNoMatchedRouteFound
+		var err error
+		switch rt.actionType {
+		case xdsresource.RouteActionUnsupported:
+			err = errRouteActionUnsupported
+		case xdsresource.RouteActionNonForwardingAction:
+			err = errRouteActionNonForwardingAction
+		default:
+			err = errNoMatchedRouteFound
+		}
+		return nil, err
```

Comment:
What about returning 
- `errNoMatchedRouteFound` error if `rt.clusters` is `nil` as it was done before this fix?
- and error as instructed in the [comment](https://github.com/grpc/grpc-go/issues/5921#issue-1528119137) only if route action type is not `RouteActionRoute`? And maybe format the error like `status.Errorf(codes.Unavailable, "received route was of type %t, require type RouteActionRoute", rt.actionType)`

Reply:
hi @tobotg, thank you for reviewing!
1. > errNoMatchedRouteFound error if rt.clusters is nil as it was done before this fix?

for `rt.clusters == nil && rt.actionType == xdsresource.RouteActionRoute`,  the default case of the switch will handle it same as before (`return nil, errNoMatchedRouteFound`) - is that satisfactory or did I misunderstand what you were saying?

2. > and error as instructed in the [comment](https://github.com/grpc/grpc-go/issues/5921#issue-1528119137) only if route action type is not RouteActionRoute

do you mean that the `rt.clusters == nil` condition is not required for routes other than `RouteActionRoute`,  and so it should be a separate check on route type only?
@easwars could you please confirm?
if this is the case then I'll rewrite the block as follows:
 ```
if rt == nil {
	return nil, errNoMatchedRouteFound
}

if rt.actionType != xdsresource.RouteActionRoute {
	return nil, status.Errorf(codes.Unavailable, "received route was of type %t, require type RouteActionRoute", rt.actionType)
}

if rt.clusters == nil {
	return nil, errNoMatchedRouteFound
}
```

Reply:
Yes, you are right in the sense that for route with action type which is not `RouteActionRoute`, the `clusters` field will be `nil`. So, we will have to handle all three cases separately, as mentioned in your reply here.